packages/components/src/PrimaryToolbar/PrimaryToolbar.js

```diff
@@ -53,11 +55,33 @@ class PrimaryToolbar extends Component {
             >
                 <ToolbarContent>
                     {
-                        (bulkSelect || filterConfig || dedicatedAction) &&
+                        (expandAll || bulkSelect || filterConfig || dedicatedAction) &&
                         <ToolbarGroup
                             className="ins-c-primary-toolbar__group-filter pf-m-spacer-md pf-m-space-items-lg"
                             variant="filter-group"
                         >
+                            {
+                                expandAll &&
+                                <ToolbarItem>
+                                    {
+                                        React.isValidElement(expandAll) ? expandAll : (
+                                            expandAll.isAllExpanded ? (
+                                                <Button variant="plain" aria-label="Collapse all" onClick={() => expandAll.onClick(false)}>
+                                                    <ToolbarExpandIconWrapper>
+                                                        <AngleDownIcon />
+                                                    </ToolbarExpandIconWrapper>
+                                                </Button>
+                                            ) : (
+                                                <Button variant="plain" aria-label="Expand all" onClick={() => expandAll.onClick(true)}>
+                                                    <ToolbarExpandIconWrapper>
+                                                        <AngleRightIcon />
+                                                    </ToolbarExpandIconWrapper>
+                                                </Button>
+                                            )
```

Comment:
No need to duplicate bunch of code here
```suggestion
                                              <Button
                                                variant="plain"
                                                aria-label={`${expandAll.isAllExpanded ? 'Collapse' : 'Expand'} all`}
                                                onClick={(e) => expandAll.onClick(e, !expandAll.isAllExpanded)}
                                              >
                                                  <ToolbarExpandIconWrapper>
                                                      {expandAll.isAllExpanded ? <AngleDownIcon /> : <AngleRightIcon />}
                                                  </ToolbarExpandIconWrapper>
                                              </Button>
```

Reply:
Nice, that's a really compact code :grinning: